index.html

```diff
@@ -1832,55 +1832,125 @@ <h2>Proof</h2>
     <h1>Core Representations</h1>
 
       <p>
-A <a>DID document</a> MUST be a single JSON object conforming to [[!RFC8259]].
-Many of the concepts in this document were introduced by example using the
-JSON-LD syntax, a format for mapping JSON data into the RDF semantic graph
-model, as defined by [[!JSON-LD]]. This section formalizes how the data model
-(described in Sections <a href="#data-model"></a> and
-<a href="#did-documents"></a>) is realized in JSON-LD.
-      </p>
-      <p>
-Although syntactic mappings are provided for JSON and JSON-LD only, applications
-and services can use any other data representation syntax, such as JXD (JSON XDI
-Data, a serialization format for the
-<a href="http://docs.oasis-open.org/xdi/xdi-core/v1.0/csd01/xdi-core-v1.0-csd01.xml">XDI graph model</a>),
+All concrete representations of a <a>DID document</a> MUST be serialized using a 
+deterministic mapping that is unambiguously able to be parsed into the data model
+defined in this specification. All serialization methods MUST define rules for 
+the bidirectional translation of a <a>DID document</a> both into and out of the
+representation in question. As a consequence, translation between any two formats
+MUST be done by first parsing the source format into a <a>DID document</a> model
+(described in Sections <a href="#data-model"></a> and <a href="#did-documents"></a>)
+and then serializing the <a>DID document</a> back out into the target format.
+      </p>
+
+      <p>
+Although syntactic mappings are provided for JSON and JSON-LD here, applications
+and services MAY use any other data representation syntax, such as
 XML, YAML, or CBOR, that is capable of expressing the data model.
       </p>
+      
+      <p>
+The production and consumption rules in this section apply to all implementations
+seeking to be fully compatible with independent implementations of the specification.
+Deployments of this specification MAY use a custom agreed-upon representation.
+See section ?? on Extensibility for more information.
+      </p>
 
     <section>
       <h2>
-Pure JSON
+JSON
       </h2>
 
+<section>
+      <h3>Production</h3>
+
         <p>
-The data model, as described in Section <a href="#data-model"></a>, can be
-encoded in Javascript Object Notation (JSON) [[RFC8259]] by mapping property
+A <a>DID document</a> MUST be a single JSON object conforming to [[!RFC8259]]. All
+top-level properties of the <a>DID document</a> MUST be represented by using the
+property name as the name of the member of the JSON object. The values of
+properties of the data model described in Section <a href="#data-model"></a>, 
+including all extensions, MUST be encoded in JSON[[RFC8259]] by mapping property
 values to JSON types as follows:
```

Comment:
There are a lot of details for the JSONLD that I am likely missing, as I am not as familiar with that specification. I repeated the text as a starting point for that greater specificity. However, it is also my understanding that all the types of fields would be accounted for by the JSONLD context, which is now required to be processed here. Remember, these are rules for processing JSONLD documents, and so we can require anyone doing so to follow JSONLD rules and expectations. I will make that more clear in the introduction text.

Reply:
So while my previous comment holds true in general for the JSON-LD section (which is incomplete), I realize now that your comment was in fact in response to the JSON section, not the JSON-LD section. In which case, this paragraph and the following list are absolutely vital and important to this specification's usefulness.

What's going on here is we are mapping between data types that we use within the DID document model to specific serialization structures within JSON. The second list tells us how to get back from JSON structures into the data types used within the DID document model. 

It is not enough to say "Just put it into JSON!" and hope for the best. Otherwise, what's to stop me from taking something that most people would do as this:

```json
{
   "foo": "bar",
   "baz": "qux"
}
```

And instead using this:

```json
[
   ["foo", "bar"],
   ["baz", "qux"]
]
```

Both are perfectly valid JSON, and could be parsed into the same underlying structure, but only one would follow the rules of this specification as I've written them out. Since this is a standard, we have the duty to be completely pedantic about how every transformation needs to work.

Reply:
> what's to stop me from taking something that most people would do as this

Well, the rule you wrote that says the top level thing MUST be a JSON object :), but I get your point. 

> Since this is a standard, we have the duty to be completely pedantic about how every transformation needs to work.

I'm fine w/ being completely pedantic for now and simplifying later if we can.

Reply:
If we can make the pedantry simpler without being less pedantic, I'm all for it. :)